_studio/shared/umc/codec/h264_dec/src/umc_h264_va_packer.cpp

```diff
@@ -702,27 +702,21 @@ void PackerVA::PackProcessingInfo(H264DecoderFrameInfo * sliceInfo)
 void PackerVA::PackQmatrix(const H264ScalingPicParams * scaling)
 {
     UMCVACompBuffer *quantBuf;
-    VAIQMatrixBufferH264* pQmatrix_H264 = (VAIQMatrixBufferH264*)m_va->GetCompBuffer(VAIQMatrixBufferType, &quantBuf, sizeof(VAIQMatrixBufferH264));
+    auto* pQmatrix_H264 = reinterpret_cast<VAIQMatrixBufferH264 *>(m_va->GetCompBuffer(VAIQMatrixBufferType, &quantBuf, sizeof(VAIQMatrixBufferH264)));
     if (!pQmatrix_H264)
         throw h264_exception(UMC_ERR_FAILED);
     quantBuf->SetDataSize(sizeof(VAIQMatrixBufferH264));
 
-    int32_t i, j;
-    //may be just use memcpy???
-    for(j = 0; j < 6; j++)
+    int32_t j;
+
+    for(j = 0; j < 6; ++j)
     {
-        for(i = 0; i < 16; i++)
-        {
-             pQmatrix_H264->ScalingList4x4[j][i] = scaling->ScalingLists4x4[j].ScalingListCoeffs[i];
-        }
+        std::copy(std::begin(scaling->ScalingLists4x4[j].ScalingListCoeffs), std::end(scaling->ScalingLists4x4[j].ScalingListCoeffs), std::begin(pQmatrix_H264->ScalingList4x4[j]));
```

Comment:
I hope you ran some tests after this change...

Reply:
Better don't change here. It is different structures. And code is increased.

Reply:
@lakulako: I am not sure what do you mean... You accepted changes and code _is_ changed...

Reply:
@dvrogozh I did manual unit testing for this change

Reply:
It is correct change for now. Now type of ScalingListCoeffs matches ScalingLists4x4[j]. But what if one of them is changed, for example to U**16**[]? Will it generate any warning?

Reply:
if there would be any change that cause conversion with possible loss of data it will generate warning, of cource